src/webutil/tableState.js

```diff
@@ -0,0 +1,244 @@
+// @flow
+
+import {type PluckFn} from "../util/sortBy";
+import sortBy from "../util/sortBy";
+import {useMemo, useState, useCallback} from "react";
+
+export const SortOrders = {
+  ASC: "asc",
+  DESC: "desc",
+};
+type SortOrderTypes = "asc" | "desc";
+
+export const DEFAULT_SORT = SortOrders.ASC;
+
+/**
+  A state manager for a React data table that provides encapsuled hooks for Sort, Pagination, and Filtering.
+  Useful for any table that needs at least one of those features, and particularly useful for managing combinations.
```

Comment:
Nit; wrap to 80 char

Reply:
wrapping all to 80, and adding a vertical ruler to my IDE for the future.